test/test_nn.py

```diff
@@ -6513,6 +6513,178 @@ def test_triplet_margin_loss_swap_no_reduce(self):
         self.assertEqual(F.triplet_margin_loss(input1, input2, input3, swap=True, reduction='none'),
                          loss_reference_fns['TripletMarginLoss'](input1, input2, input3, swap=True, reduction='none'))
 
+    def test_triplet_margin_loss_with_distance_parity(self):
+        anchor = torch.randn(5, 10, requires_grad=True)
+        positive = torch.randn(5, 10, requires_grad=True)
+        negative = torch.randn(5, 10, requires_grad=True)
+        self.assertTrue(gradcheck(lambda a, p, n: F.triplet_margin_loss_with_distance(
+            a, p, n), (anchor, positive, negative)))
+        self.assertEqual(F.triplet_margin_loss_with_distance(anchor, positive, negative),
+                         F.triplet_margin_loss(anchor, positive, negative))
+
+    def test_triplet_margin_loss_with_distance(self):
+        anchor = torch.randn(5, 10, requires_grad=True)
+        positive = torch.randn(5, 10, requires_grad=True)
+        negative = torch.randn(5, 10, requires_grad=True)
+        distance_fn = nn.PairwiseDistance()
+        self.assertTrue(gradcheck(lambda a, p, n: F.triplet_margin_loss_with_distance(
```

Comment:
This and the following test(s) seem like they could be combined by iterating over distance functions or parameter packs.

Reply:
For example:

```
for d, swap in product(distance_functions, (True, False)):
   ... 
```